tests/simulation/test_download.py

```diff
@@ -154,3 +156,16 @@ def test_match_filetypes():
     ]
 
     assert Downloader.match_filetypes(manifest) == expected_filetypes
+
+
+def test_compare_checksums():
+    parent_dir = pathlib.Path(__file__).parent.parent.absolute()
+    file_path = str(parent_dir / "mock_data" / "calib000000.txt")
+    checksum_path = str(parent_dir / "mock_data" / "checksum_calib000000.txt")
+
+    raise_error = False
+    try:
```

Comment:
Should use pytest to assert raises like [this](https://docs.pytest.org/en/3.0.1/assert.html#assertions-about-expected-exceptions)

Reply:
Done.

Reply:
Why do you need a `no_raises` method? If the checksum passes, it should return True/False to validate it passed. From the method called `compare_checksums` it implies the method will return the checksum results (whether it passes/failed)

Reply:
But if we just return True/False, every time we do the checksum, we need to raise error:
```
if not compare_checksums(
        file_path=file_dest_path, checksum_path=checksum_dest_path
    ):
        raise DownloadError(
            f"Invalid hash value."
            f"There are some errors during the dataset download. "
            f"Please download it again."
        )
```